src/gc_roots.cpp

```diff
@@ -115,6 +189,18 @@ GcTag *pointerToGcTag(jlong tagPtr) {
     return (GcTag *) (ptrdiff_t) (void *) tagPtr;
 }
 
+GcTag *createWeakSoftReferenceClassTag() {
+    return new WeakSoftReferenceClassTag();
+}
+
+GcTag *createGcTag(jlong classTag) {
+    return new GcTag(classTag != 0 && pointerToGcTag(classTag)->isWeakSoftReachable());
```

Comment:
Can we have tag balance like in `src/objects_size.cpp:9`. At some point, we would need to generalize it, but for now simple counting seems enough.